src/common/WorkQueue.cc

```diff
@@ -122,6 +122,8 @@ void ThreadPool::worker(WorkThread *wt)
 	  processing++;
 	  ldout(cct,12) << "worker wq " << wq->name << " start processing " << item
 			<< " (" << processing << " active)" << dendl;
+          assert(strlen(wq->thread_name) < 16);
+          pthread_setname_np(wt->get_thread_id(), wq->thread_name);
```

Comment:
Oh sorry, I forgot before that threads belong to the pool, not the work queue.  The thead pool has a name as well; I'd change *that* constructor to take 2 name args (and drop the WorkQueue changes).  Or, just assert that the existing threadpool name is short--that's probably simpler.

I don't think it's a good idea to introduce any per-work-item overhead (like setting the thread name).  I don't think it'll make much difference to someone debugging... usually identifying the threadpool tells you which work it is anyway (they aren't frequently shared).